src/basic/cpu-set-util.c

```diff
@@ -60,49 +60,54 @@ cpu_set_t* cpu_set_malloc(unsigned *ncpus) {
         }
 }
 
-int parse_cpu_set_and_warn(
+static int parse_cpu_set_internal(
                 const char *rvalue,
                 cpu_set_t **cpu_set,
+                bool warn,
                 const char *unit,
                 const char *filename,
                 unsigned line,
                 const char *lvalue) {
 
-        const char *whole_rvalue = rvalue;
         _cleanup_cpu_free_ cpu_set_t *c = NULL;
+        const char *p = rvalue;
         unsigned ncpus = 0;
 
-        assert(lvalue);
         assert(rvalue);
 
         for (;;) {
                 _cleanup_free_ char *word = NULL;
                 unsigned cpu, cpu_lower, cpu_upper;
                 int r;
 
-                r = extract_first_word(&rvalue, &word, WHITESPACE ",", EXTRACT_QUOTES);
+                r = extract_first_word(&p, &word, WHITESPACE ",", EXTRACT_QUOTES);
+                if (r == -ENOMEM)
+                        return warn ? log_oom() : -ENOMEM;
                 if (r < 0)
-                        return log_syntax(unit, LOG_ERR, filename, line, r, "Invalid value for %s: %s", lvalue, whole_rvalue);
+                        return warn ? log_syntax(unit, LOG_ERR, filename, line, r, "Invalid value for %s: %s", lvalue, rvalue) : r;
```

Comment:
hmm, maybe just do this instead:

```c
return log_syntax(unit, warn ? LOG_WARNING : LOG_DEBUG, …)
```

i.e. always log, but downgrade it to debug (which is always OK) if the caller didn't ask for proper logging...

Reply:
Hmm. parse_cpuset() is also used in systemd-run. In that case, filename and line is meaningless and thus log message become confusing. Even if the message is logged in debug, I think we should not make confusing log.

Reply:
good point

Reply:
I think the ternary operator is not a great fit here. Why not make it a full `if`?